lib/assets/javascripts/turbograft/turbolinks.coffee

```diff
@@ -56,14 +56,204 @@ removeNode = (node) ->
   removedNode = node.parentNode.removeChild(node)
   triggerEvent('page:after-node-removed', removedNode)
 
+updateHeadOrRefresh = (url, upstreamDocument, callback) ->
```

Comment:
`Or` methods are usually a code smell.  Most of this doesn't look like it has to be done async.  There should be:
* A method that collates information for before/after compatibility
* Logic in `loadPage` that decides whether to do a full refresh/inline update
* A method that does the inline update

It makes not sense to kick off `updateLinkTags`, then start a full page refresh immediately afterwards.